lib/order.rb

```diff
@@ -0,0 +1,58 @@
+
+require 'csv'
+require_relative 'customer'
+
+class Order
+  attr_reader :id, :products, :customer, :fulfillment_status
+  attr_writer :products, :customer, :fulfillment_status
+
+  @@valid_statuses = %i[pending paid processing shipped complete]
+  @@orders = []
+
+  def initialize(id, products, customer, fulfillment_status = :pending)
+    @id = id
+    @products = products
+    @customer = customer
+    @fulfillment_status = fulfillment_status
+    valid(fulfillment_status)
+  end
+
+  def valid(fulfillment_status)
+    unless @@valid_statuses.include?(fulfillment_status)
+      raise ArgumentError
+    end
+  end
+
+  def total
+    tax = products.values.sum * 0.075
```

Comment:
It's better to only run `.sum` once per method invocation.  Otherwise it's a bit wasteful.

So something like:

```ruby
subtotal = products.values.sum
tax = subtotal * 0.075
total = tax + subtotal
return total.round(2)
```